src/gpio.rs

```diff
@@ -590,6 +590,7 @@ macro_rules! impl_input_only {
 macro_rules! impl_input_output {
     ($pxi:ident: $pin:expr) => {
         impl_input_base!($pxi: $pin);
+        impl_pull!($pxi: Unknown);
```

Comment:
Why? Shouldn't we enable the `Pull` trait only for pins in `Input/InputOutput/SubscribedInput/Analog*` modes?

Reply:
Was implementing the suggestion here: https://github.com/esp-rs/esp-idf-hal/issues/55#issuecomment-1084348972

Happy to remove it though.

Reply:
I don't think `Pull` is implemented for `Output`, as you originally stated?
I think it is only implemented for `Input*` and `SubscribedInput*` (the last one since today, as it makes sense), and I suggest we *only additionally* implement it for the ADC modes. So yeah, I'll remove that line. I'm anyway pushing for a new build. :)